public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.test.tsx

```diff
@@ -155,4 +137,30 @@ describe('VariableEditor:', () => {
       });
     });
   });
+
+  describe('predefined queries:', () => {
+    it('should show the new query types if feature gate is enabled', async () => {
+      grafanaRuntime.config.featureToggles.azTemplateVars = true;
+      render(<VariableEditor {...defaultProps} />);
+      openMenu(screen.getByLabelText('select query type'));
+      await waitFor(() => expect(screen.getByText('Subscriptions')).toBeInTheDocument());
+    });
+
+    it('should not show the new query types if feature gate is disabled', async () => {
+      grafanaRuntime.config.featureToggles.azTemplateVars = false;
+      render(<VariableEditor {...defaultProps} />);
+      openMenu(screen.getByLabelText('select query type'));
+      await waitFor(() => expect(screen.queryByText('Subscriptions')).not.toBeInTheDocument());
+    });
+
+    it('should run the query if requesting subscriptions', async () => {
+      grafanaRuntime.config.featureToggles.azTemplateVars = true;
+      const onChange = jest.fn();
+      render(<VariableEditor {...defaultProps} onChange={onChange} />);
+      openMenu(screen.getByLabelText('select query type'));
+      screen.getByText('Subscriptions').click();
+      await waitFor(() => expect(screen.getByText('Subscriptions')).toBeInTheDocument());
```

Comment:
A _tiny_ nit: this seems redundant given line 146 above.

Reply:
it's necessary because I need to wait for the `click` to take effect. If I don't do this, the test ends too soon and I get the warning `Warning: An update to VariableEditor inside a test was not wrapped in act(...).`